### PR TITLE
Fix duplicate symbols in NCO

### LINKED ISSUE #390: Duplicate symbols in NCO build
Need to add `CPPFLAGS="-DHAVE_NETCDF4_H"` to indicate we're using external NetCDF-4.

See http://nco.sourceforge.net/build_hints.shtml

### PR DESCRIPTION
**Pull request by kgerheiser**
Fix #390

### CODE REVIEW

**Review by Hang-Lei-NOAA — APPROVED** [review]

**Review by climbfuji — APPROVED** [review]

File: libs/build_nco.sh [hunk 1]
```diff
@@ -89,6 +89,10 @@ export LIBS="${PNETCDF_LIBS:-} ${NETCDF_LIBS:-} ${HDF5_LIBS:-} ${EXTRA_LIBS:-}"
 [[ -d build ]] && rm -rf build
 mkdir -p build && cd build
 
+# Prevents duplicate symbols
+# See http://nco.sourceforge.net/build_hints.shtml
+CPPFLAGS="-DHAVE_NETCDF4_H"
```
Comment (aerorahul) [inline libs/build_nco.sh]: Not that we will likely support installing netcdf version 3, but if we did, do we check for the netcdf version with this:
```
❯❯❯ nc-config --version
netCDF 4.7.4
```

**Review by aerorahul — APPROVED** [review]

File: libs/build_nco.sh (hunk 1 above)
Comment (edwardhartnett) [reply libs/build_nco.sh]: There is never a need to install an older version of netCDF. All netcdf libraries are fully backward compatible. NetCDF-3 code will run exactly the same, and produce the exact same output, with the latest version of netCDF.

Comment (aerorahul) [reply libs/build_nco.sh]: Presently, there are applications in the production suite that rely on netCDF 3.6.3 and link with it.

Comment (kgerheiser) [reply libs/build_nco.sh]: Sure, why not. Added a check for NetCDF4.

**Comment by kgerheiser** [comment]
That fix worked for me on an Ubuntu machine, but on Orion I'm still getting duplicate symbols.

```
In function `nc_open_mem':
dfile.c:(.text+0x820): multiple definition of `nc_open_mem'
./.libs/libnco.a(nco_netcdf.o):nco_netcdf.c:(.text+0x9d0): first defined here
/work/noaa/stmp/gkyle/hpc-stack/hpc-stack-gfsv16.2/install/intel-2018.4//netcdf/4.7.4/lib/libnetcdf.a(libdispatch_la-dfile.o): In function `nc_open_mem':
dfile.c:(.text+0x820): multiple definition of `nc_open_mem'
./.libs/libnco.a(nco_netcdf.o):nco_netcdf.c:(.text+0x9d0): first defined here
/work/noaa/stmp/gkyle/hpc-stack/hpc-stack-gfsv16.2/install/intel-2018.4//netcdf/4.7.4/lib/libnetcdf.a(libdispatch_la-dfile.o)/work/noaa/stmp/gkyle/hpc-stack/hpc-stack-gfsv16.2/install/intel-2018.4//netcdf/4.7.4/lib/libnetcdf.a(libdispatch_la-dfile.o): In function `: In function `nc_open_memnc_open_mem':
':
dfile.c:dfile.c:((.text.text+0x+0x820820)): multiple definition of `: multiple definition of `nc_open_memnc_open_mem'
'
./.libs/libnco.a(nco_netcdf.o)./.libs/libnco.a(nco_netcdf.o)::nco_netcdf.c:nco_netcdf.c:((.text.text+0x+0x9d09d0)): first defined here
: first defined here
/work/noaa/stmp/gkyle/hpc-stack/hpc-stack-gfsv16.2/install/intel-2018.4//netcdf/4.7.4/lib/libnetcdf.a(libdispatch_la-dfile.o): In function `nc_open_mem':
dfile.c:(.text+0x820): multiple definition of `nc_open_mem'
./.libs/libnco.a(nco_netcdf.o):nco_netcdf.c:(.text+0x9d0): first defined here
/work/noaa/stmp/gkyle/hpc-stack/hpc-stack-gfsv16.2/install/intel-2018.4//netcdf/4.7.4/lib/libnetcdf.a(libdispatch_la-dfile.o): In function `nc_open_mem':
dfile.c:(.text+0x820): multiple definition of `nc_open_mem'
./.libs/libnco.a(nco_netcdf.o):nco_netcdf.c:(.text+0x9d0): first defined here
/work/noaa/stmp/gkyle/hpc-stack/hpc-stack-gfsv16.2/install/intel-2018.4//netcdf/4.7.4/lib/libnetcdf.a(libdispatch_la-dfile.o): In function `nc_open_mem':
dfile.c:(.text+0x820): multiple definition of `nc_open_mem'
./.libs/libnco.a(nco_netcdf.o):nco_netcdf.c:(.text+0x9d0): first defined here
/work/noaa/stmp/gkyle/hpc-stack/hpc-stack-gfsv16.2/install/intel-2018.4//netcdf/4.7.4/lib/libnetcdf.a(libdispatch_la-dfile.o): In function `nc_open_mem':
dfile.c:(.text+0x820): multiple definition of `nc_open_mem'
./.libs/libnco.a(nco_netcdf.o):nco_netcdf.c:(.text+0x9d0): first defined here
```

**Comment by kgerheiser** [comment]
Using the latest version of NCO fixes the problem.

**Comment by kgerheiser** [comment]
I'm going to close this and combine with #392

**Pull request closed without merging**